rfcs/rfc00XX.md

```diff
@@ -0,0 +1,257 @@
+count?), then any attempt to import a function not designated for export should
+be a fatal error:
+
+    $ perl -MMy::Module=whee -E 1
+    "whee" is not exported by the My::Module module
+    Can't continue after import errors at -e line 0.
+    BEGIN failed--compilation aborted.
+
+For the following, we'll use the following stub example of an `import()`
+method:
+
+    package My::Module {
+        use v5.38;
+
+        sub import ($class, @args) {
+            ...
+        }
+        sub bar  :export            {...} # @EXPORT_OK
+        sub baz  :export(:strings)  {...} # %EXPORT_TAGS
+        sub bay  :export(:strings)  {...} # %EXPORT_TAGS
+        sub quux :export(:ALWAYS)   {...} # @EXPORT
+        sub whee                    {...} # not exported
+    }
+
+If an `import()` method is defined, it's possible that it might have special
+importing functionality provided, but we will _still_ allowing importing of
+functions with `:export` tags. We pass the import arguments unchanged to the
+`import()` method (in `@args`, above) and the module author will be
+responsible for filtering those manually. This seems unfortunate, but if
+someone uses `use v5XX`, their import mechanism shouldn't suddenly break.
+
+As a counter-argument, it's unlikely someone will be using `:export`
+attributes, so we could simply filter out all subnames so they don't populate
+`@args`, or filter out all subnames tagged with `:export`.
+
+For the above, that means that these are problematic:
+
+    use My::Module ':srtings';
+    use My::Module 'whee';
+
+For `:srtings`, that's almost certainly misspelled. For `whee`, that's not
+supposed to be exported, but perhaps the module author wants to allow that in
+the argument list.
+
+To side-step this, we could create an alternate to the `import()` method,
+perhaps as something analogous to a phaser, but taking arguments, or a new
+`IMPORT` method. Bike-shedding may now commence!
+
+## Security Implications
+
+None anticipated, but by having the imports being lexical, it means outside
+code might have a harder time replacing these functions.
+
+## Examples
+
+    package My::Module {
+        use v5.38;
+
+        sub bar  :export            {...} # @EXPORT_OK
+        sub baz  :export(:strings)  {...} # %EXPORT_TAGS
+        sub bay  :export(:strings)  {...} # %EXPORT_TAGS
+        sub quux :export(:ALWAYS)   {...} # @EXPORT
+        sub whee                    {...} # not exported
+    }
+
+    use My::Package 'bar';     # imports bar() and quux
+
+    use My::Package ':strings' # imports baz, bay, and quux
+
+    use My::Package;           # imports quux
+
+    use My::Package ();        # imports nothing
+
+    use My::Package qw/bar :strings -quux/; # don't import quux
+
+    use My::Module 'whee'; # fatal error
+
+## Prototype Implementation
+
+None that I'm aware of, though
+[Perl6::Export::Attrs](https://metacpan.org/pod/Perl6::Export::Attrs) seems
+the closest.
+
+## Future Scope
```

Comment:
Another possible future idea: importing as another name. Since they're lexical imports and not added to the externally-visible namespace of the module that does the importing, it should be safe to rename them. Maybe support some notation at import time, where an importer can import a function as a different name to avoid clashes or for general clarity.